automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSuite.scala

```diff
@@ -243,6 +249,46 @@ trait NewBillingProjectAndWorkspaceBeforeAndAfterAll extends BillingProjectUtils
     } else IO.unit
 }
 
+final case class AzureBillingProjectName(value: String) extends AnyVal
+trait AzureBillingBeforeAndAfter extends FixtureAnyFreeSpecLike with BeforeAndAfterAll {
+  this: TestSuite =>
+  override type FixtureParam = AzureBillingProjectName
+
+  val azureBillingProjectKey = "leonardo.azureBillingProject"
+
+  implicit val (authToken, authorization) = getAuthTokenAndAuthorization(Hermione)
+  implicit val azureManagedAppCoordinates = AzureManagedAppCoordinates(
+    UUID.fromString("fad90753-2022-4456-9b0a-c7e5b934e408"),
+    UUID.fromString("f557c728-871d-408c-a28b-eb6b2141a087"),
+    "staticTestingMrg",
+    Some(UUID.fromString("f41c1a97-179b-4a18-9615-5214d79ba600"))
+  )
+
+  // Needed for wb-libs
+  implicit val accessToken = Hermione.authToken().unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+
+  override def withFixture(test: OneArgTest): Outcome = {
+    def runTestAndCheckOutcome(projectName: AzureBillingProjectName) = {
+      super.withFixture(test.toNoArgTest(projectName))
+    }
+
+    sys.props.get(azureBillingProjectKey) match {
+      case None => throw new RuntimeException(s"${azureBillingProjectKey} system property is not set, indicating a problem with azure billing project setup")
+      case Some(projectName) => runTestAndCheckOutcome(AzureBillingProjectName(projectName))
+    }
+  }
+
+
+  override def beforeAll(): Unit = {
+    // hardcode this if you want to use a static azure billing project
+    //val projectName = "tmp-billing-project-beddf71a74"
```

Comment:
I'm going to keep these here for now since it makes local testing easy, and the proper places are documented in the README. Once these tests are 'stable', we can change it to something like that